shared/src/commonMain/kotlin/org/hyperskill/app/step/domain/model/CommentStatisticsEntry.kt

```diff
@@ -6,7 +6,7 @@ import kotlinx.serialization.Serializable
 @Serializable
 data class CommentStatisticsEntry(
     @SerialName("thread")
-    val thread: CommentThread,
+    val thread: CommentThread? = null,
```

Comment:
Backend can send an unknown value in the `thread` field. We should handle it properly with default value. This is why `thread` should be nullable. The alternative for it is a default enum value like `UNKNOWN`.

Reply:
> Backend can send an unknown value in the thread field. We should handle it properly with default value. This is why thread should be nullable.

Yes

> The alternative for it is a default enum value like UNKNOWN.

I think a null is better in case of backend returns an unknown thread type for the mobile client. Nullable forces us to unwrap it before using it, but with a non-nullable value we can forget to check for the `UNKNOWN` type.

Reply:
In my opinion, the `UNKNOWN` enum value is better because it presents human readable value and clearly shows the state of the `CommentThread`.  We just need to check all the values in the `when` expression.

Reply:
@XanderZhu @ivan-magda Let's discuss that on next PBR, now we have different approaches in the project. We will choose the one and create task to technical debt